- Cleaned up all references

olsonpm opened the pull request:
- Added an instructions file on how to build/debug the project from a fresh pull

olsonpm committed:
- Cleaned up all references

 - Added an instructions file on how to build/debug the project from a fresh pull
 -

olsonpm commented:
So I've done a pull request once before, I'm more used to svn though I'm starting to like git a lot.  Anyway - the one change I wasn't sure how to go about were the external 'Common' files that exist in the np++ codebase.  Due to np++ not being tracked in git, we couldn't create an external reference (I forget the term git gives it), so are we stuck with hard copies of the files?  I didn't spend the time to figure out those references to be as clean as possible, so I feel the committed solution will work.

The only other note I should add is that this commit is purely to clean up the existing repo.  It doesn't do any functionality changes.

bruderstein commented on src/stdafx.cpp:
```diff
@@ -2,7 +2,7 @@
 // stdafx.obj will contain the pre-compiled type information
 
-#include "stdafx.h"
+#include <precompiledHeaders.h>
```
These should all be #include "precompiledHeaders.h"  (not <precompiledHeaders.h>)

Can you fix these?

olsonpm replied:
Sure, but then we will need two separate standard header files (one in the
common dir, and one in src).  Is that best practice?  I will do some
reading on that later today to make sure i understand correctly.

On Saturday, May 3, 2014, Dave Brotherstone notifications@github.com
wrote:

> In src/stdafx.cpp:
> 
> > @@ -2,7 +2,7 @@
> >  // BufferSwitcher.pch will be the pre-compiled header
> >  // stdafx.obj will contain the pre-compiled type information
> > 
> > -#include "stdafx.h"
> > +#include <precompiledHeaders.h>
> 
> These should all be #include "precompiledHeaders.h" (not )
> 
> Can you fix these?
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/davegb3/FileSwitcher/pull/1/files#r12256209
> .

bruderstein replied:
Sharing precompiledHeaders between projects is bad practice - this isn't doing that, but it's on the way as you've got two sort-of sub projects using the same header via the include path.  I've found using a header from the include path (ie. with <header>), can lead to some things being built unnecessarily, or some files always rebuilding.  

The common directory shouldn't really share any files with the src directory, but likewise it doesn't really make sense to put the precompiled header in the common directory, then reference it from source, as that would mean your project specific general headers would go in as "common".  I don't think having two files works, as you'd need two lots of precompiled header output, which I don't think you can have for a single project.  Splitting this into two projects would be the right way to go, but that's probably quite a bit of effort.  I would probably pragmatically say put the precompiledHeader.h in src, and for the common CPPs, just use #include "../src/precompiledHeader.h".  You might have to set the name of the precompiled header in the properties of the common cpp files too.

olsonpm replied:
That makes too much sense. Thanks much for the explanation.

On Saturday, May 3, 2014, Dave Brotherstone notifications@github.com
wrote:

> In src/stdafx.cpp:
> 
> > @@ -2,7 +2,7 @@
> >  // BufferSwitcher.pch will be the pre-compiled header
> >  // stdafx.obj will contain the pre-compiled type information
> > 
> > -#include "stdafx.h"
> > +#include <precompiledHeaders.h>
> 
> Sharing precompiledHeaders between projects is bad practice - this isn't
> doing that, but it's on the way as you've got two sort-of sub projects
> using the same header via the include path. I've found using a header from
> the include path (ie. with ), can lead to some things being built
> unnecessarily, or some files always rebuilding.
> 
> The common directory shouldn't really share any files with the src
> directory, but likewise it doesn't really make sense to put the precompiled
> header in the common directory, then reference it from source, as that
> would mean your project specific general headers would go in as "common". I
> don't think having two files works, as you'd need two lots of precompiled
> header output, which I don't think you can have for a single project.
> Splitting this into two projects would be the right way to go, but that's
> probably quite a bit of effort. I would probably pragmatically say put the
> precompiledHeader.h in src, and for the common CPPs, just use #include
> "../src/precompiledHeader.h". You might have to set the name of the
> precompiled header in the properties of the common cpp files too.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/davegb3/FileSwitcher/pull/1/files#r12258113
> .

olsonpm committed:
Rename common/precompiledHeaders.cpp to src/precompiledHeaders.cpp

olsonpm committed:
Rename common/precompiledHeaders.h to src/precompiledHeaders.h

olsonpm committed:
Fix precompiled header file references

Moved to src dir, changed from <> to "" convention

olsonpm committed:
Precompiled header final fixes

Missed a couple project settings to make sure the build completes.  Had
to change the resource paths too for some reason (was working fine
before, but wouldn't compile this time around).

olsonpm committed:
configDialog bug and gitignore change

After testing, I found an incorrectly modified line in my initial
commit.  Fixed this, and added *.sdf filter to .gitignore

olsonpm commented:
Alright - everything should be good.  The project builds and I briefly tested the functionality.  I don't intend on this being released to np++, so I'll hold on the thorough testing until my added features are developer done.

olsonpm committed:
*.sdf

*.aps
 - committed one last time before they will be ignored

.gitignore
 - fix to correctly ignore *.sdf & *.aps files

FileSwitcher
 .sln
 .vcproj
 .vcxproj
 .filters
 - Necessary project file changes

common\StaticDialog.cpp
 - fixed bug where switcher dialog was staying on top of other windows as well (e.g. chrome).  This could be because I have a windows program that creates virtual desktops - not sure, but the reason I say that is because documentation suggests hwnd_top shouldn't behave as I was experiencing.

how-to-build-project.txt
 - Modified order, no real changes

questionMark.bmp
 - Added tooltip icon for the config dialog

BufferSwitcher.h
 - Wasn't being used

Compare.h
 - needed tchar.h include

ConfigDialog.cpp
 .h
 - now includes 'globals.h' which, though not comprehensive, has most of the globals which were previously defined in FileSwitcher.h
 - White space changes / spelling fixes / cleaning some code to either be more readable or take less lines
 - Added wildcard
 - Added feature of configured context, along with settings enabled/disabled depending.
 - Along with configured context, enabled configurable number of traversed files and displayed files to save from np++ crashes.
 - Changed reverse sort order from computer math magic to a boolean (I was getting buggy behavior with this feature either after adding 'NONE' to the enum or by adding some other feature, hence the change)
 - Got rid of duplicate IDC value for having a dialog for ctrl+tab.  Also changed negative worded variables to positive.
 - Fixed some enabling/disabling control logic (e.g. revert alt+tab was getting enabled dependent upon the index sort override)
 - Added friendly error checking upon clicking [ok]
 - Added tooltips for new features
 - Made sure context files were only being loaded into memory if the context changed, or context changed from open tabs -> configured folder.

EditFile.cpp
 .h
 - Added "HasMultipleViews" which tells you whether np++ has multiple views open.  This addresses a bug in np++ where the getopendocuments message returns a file "new  0" taking up view 1, and all other files taking up view 2 - even though no such file exists.
 - Modified returned indexes and views throughout the code to address the above bug as well
 - Got rid of void* scintillaDoc variables/functions due to it not being used throughout the codebase.

FileListView.cpp
 .h
 - Set imagelist to none for configured directory searches, as they don't apply
 - Got rid of some redundant, but unharmful code.
 - Added ability to get the currently selected "simpleFile" in the switchDialog's listview
 - Added Sorting capabilities for SimpleFileInfo
 - Added code to ensure the view was displaying SimpleFiles correctly
 - With ReverseSortOrder now being a boolean, I added ActiveReverseSortOrder as a complement to the default setting
 - Got rid of View and Index columns for SimpleFiles
 - Added function to get the current number of columns in the listview

FileSwitcher.cpp
 - Got rid of scintilla doc vars&fxns for a reason stated above
 - Added new features to init & save settings
 - Added function to recursively traverse the configured path and load all the filenames+paths into memory, with a check for going past the configured maximum amount
 - A lot of whitespace changes / deleted commented out code
 - Added a couple global functions isANumber and CreateTooltip

FileSwitcher.h
Globals.h
 .cpp
 - Moved almost all of this code over to a file "Globals.h"
 - Kept all the same includes in case other files were dependent upon it.
 - Added struct SimpleFileInfo which is just a container for fileName and filePath
 - Added to options_t due to added features
 - Added a boolean to signify when the switchdialog was showing due to ctrl+tab being pressed.

stdafx.cpp
 .h
PrecompiledHeaders.h
 - self explanatory

SearchOptions.h
 - Got rid of REVERSE_SORT_ORDER
 - Added SEARCHFLAG_INCLUDEWILDCARD
 - Added NONE to struct SortOrder
 - Replaced by PrecompiledHeaders.

SwitchDialog.cpp
 .h
 - Added separate doDialog for simple files as opposed to currently open EditFiles
 - Got rid of redundant calls to search/sort
 - Fixed reset sorting vs remembering last sort bugs
 - Again, HWND_TOP -> _hParent due to bizarre TOPMOST behavior
 - Added setupColumnWidths fxn for simplefiles
 - Added/Modified code to ensure ctrl+tab features worked as expected after adding new functionality
 - Added code to make sure correct search algorithms were being called dependent upon configured/non configured search context
 - Added code to open a simple file once selected
 - Updated search algorithms to be regex based instead of traversing through characters

TestingProcedure.txt
 - Added somewhat comprehensive directions to test in the future.  Saves a man from thinking about it.

dirent.h
 - A useful third party tool I used to traverse the file system.
